### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by coreyhulen**

### CODE REVIEW

File: api/web_hub.go [hunk 1]
```diff
@@ -115,84 +110,23 @@ func (h *Hub) Start() {
...
-					if shouldSendEvent(webCon, msg) {
-						select {
-						case webCon.Send <- msg:
-						default:
-							close(webCon.Send)
-							delete(h.connections, webCon)
-						}
+					if webCon.Broadcast(msg) {
+						delete(h.connections, webCon)
+						webCon.Close()
```
Comment (jwilander) [inline api/web_hub.go]: This seems a bit weird, if the message broadcasts successfully `webCon.Broadcast` returns false. Shouldn't it be the other way around, or am I missing something?

Comment (coreyhulen) [inline api/web_hub.go]: I've removed this.  Don't think it was needed, and change it to a blocking call.

**Comment by enahum** [comment]
Oh! I think this is failing cause of the recent merge to master and ent, and you should probably rebase and fix the conflicts

File: api/team.go [hunk 2]
```diff
@@ -304,8 +304,8 @@ func JoinUserToTeam(team *model.Team, user *model.User) *model.AppError {
 	RemoveAllSessionsForUserId(user.Id)
 	InvalidateCacheForUser(user.Id)
 
-	// This message goes to every channel, so the channelId is irrelevant
-	go Publish(model.NewWebSocketEvent("", "", user.Id, model.WEBSOCKET_EVENT_NEW_USER))
+	// This message goes to everyone, so the teamId, channelId and userId are irrelevant
+	go Publish(model.NewWebSocketEvent(model.WEBSOCKET_EVENT_NEW_USER, "", "", "", nil))
```
Comment (jwilander) [inline api/team.go]: Isn't the user ID needed by the client to know which user to fetch?

File: api/user.go [hunk 3]
```diff
@@ -271,8 +271,8 @@ func CreateUser(user *model.User) (*model.User, *model.AppError) {
 
 		ruser.Sanitize(map[string]bool{})
 
-		// This message goes to every channel, so the channelId is irrelevant
-		go Publish(model.NewWebSocketEvent("", "", ruser.Id, model.WEBSOCKET_EVENT_NEW_USER))
+		// This message goes to everyone, so the teamId, channelId and userId are irrelevant
+		go Publish(model.NewWebSocketEvent(model.WEBSOCKET_EVENT_NEW_USER, "", "", "", nil))
```
Comment (jwilander) [inline api/user.go]: Same as above

File: api/team.go (hunk 2 above)
Comment (enahum) [inline api/team.go]: yeap, it should be included in the Data property, I'll submit a fix for both

File: config/config.json [hunk 4]
```diff
@@ -117,8 +117,9 @@
         "EmailBatchingInterval": 30
     },
     "RateLimitSettings": {
-        "EnableRateLimiter": true,
+        "Enable": false,
```
Comment (crspeller) [inline config/config.json]: Some rate limit changes we don't want.

Comment (coreyhulen) [inline config/config.json]: @crspeller what should this be? If I remove the entire RateLimitSetting the server sets the defaults as

```
    "RateLimitSettings": {
        "Enable": false,
        "PerSec": 10,
        "MaxBurst": 100,
        "MemoryStoreSize": 10000,
        "VaryByRemoteAddr": true,
        "VaryByHeader": ""
    },
```

**Comment by coreyhulen** [comment]
InvalidateCacheForChannel is no longer need, but I cannot remove it since it's in the enterprise repo, so I just commented it out for now.

**Pull request merged**